ob1k-cache/src/main/java/com/outbrain/ob1k/cache/RefreshLoadingCacheDelegate.java

```diff
@@ -0,0 +1,303 @@
+package com.outbrain.ob1k.cache;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.collect.Iterables;
+import com.outbrain.ob1k.concurrent.ComposableFuture;
+import com.outbrain.swinfra.metrics.api.Counter;
+import com.outbrain.swinfra.metrics.api.MetricFactory;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+/**
+ * A wrapper of LoadingCacheDelegate that adds the ability to refresh cache values in the configured interval on access
+ */
+public class RefreshLoadingCacheDelegate<K, V> implements TypedCache<K, V> {
```

Comment:
We should think about the design here, because it feels like mostly a copy of LoadingCacheDelegate